crates/derive/src/stages/channel_reader.rs

```diff
@@ -35,7 +35,15 @@ pub trait ChannelReaderProvider {
     async fn next_data(&mut self) -> PipelineResult<Option<Bytes>>;
 }
 
-/// [ChannelReader] is a stateful stage that does the following:
```

Comment:
lmfao, I know exactly what I was thinking when this happened
![image](https://github.com/user-attachments/assets/8064c685-b2fb-4ba5-8e01-b900fa7b05a2)